Objects/typeobject.c

```diff
@@ -6653,7 +6653,9 @@ type_add_method(PyTypeObject *type, PyMethodDef *meth)
         descr = PyDescr_NewClassMethod(type, meth);
     }
     else if (meth->ml_flags & METH_STATIC) {
-        PyObject *cfunc = PyCFunction_NewEx(meth, (PyObject*)type, NULL);
+        PyObject *mod = PyObject_GetAttr((PyObject*)type, &_Py_ID(__module__));
```

Comment:
This might theoretically return `NULL` and raise `AttributeError`.

Reply:
Hmm, thanks.  Probably we could just clear the error indicator on ``mod == NULL``.

Reply:
Also take a look at `GetOptionalAttr`

Reply:
I think we want ignore here not just AttributeError's.  I did similar fix for class methods and got a failure in the test.test_types.UnionTests.test_or_type_operator_with_bad_module().

Reply:
It is better to use `type_module()` for consistency with the rest of this file. It will not give a surprise if Exceptions other than AttributeError should not be ignored.